src/Rules/Drupal/GlobalDrupalDependencyInjectionRule.php

```diff
@@ -47,7 +47,7 @@ public function processNode(Node $node, Scope $scope): array
         ];
         $classReflection = $scopeClassReflection->getNativeReflection();
         foreach ($whitelist as $item) {
-            if ($classReflection->implementsInterface($item)) {
+            if (interface_exists($item) && $classReflection->implementsInterface($item)) {
```

Comment:
I think we can work around this by getting the interfaces names versus checking if it implements the interface. Working on a new PR.